Description: Fix minion job returns handling for batches
Bug: https://github.com/saltstack/salt/issues/23047
Author: Ariel T. Glenn <ariel@wikimedia.org>
Forwarded: yes
Applied-Upstream: https://github.com/saltstack/salt/commit/b351af042d94f2f4a693a69655134b61400f3de6
--- a/salt/cli/batch.py
+++ b/salt/cli/batch.py
@@ -135,12 +135,14 @@
             parts = {}
 
             # see if we found more minions
-            for ping_ret in self.ping_gen:
-                if ping_ret is None:
-                    break
-                if ping_ret not in self.minions:
-                    self.minions.append(ping_ret)
-                    to_run.append(ping_ret)
+            if self.ping_gen is not None:
+                for ping_ret in self.ping_gen:
+                    if ping_ret is None:
+                        break
+                    for minion_name in ping_ret.keys():
+                        if minion_name not in self.minions:
+                            self.minions.append(minion_name)
+                            to_run.append(minion_name)
 
             for queue in iters:
                 try:
